webui/react/src/pages/Templates/TemplateViewModal.tsx

```diff
@@ -0,0 +1,52 @@
+import Avatar from 'hew/Avatar';
+import CodeEditor from 'hew/CodeEditor';
+import { Modal } from 'hew/Modal';
+import { Label } from 'hew/Typography';
+import yaml from 'js-yaml';
+import { useMemo } from 'react';
+
+import { NavigationItem } from 'components/NavigationSideBar';
+import { paths } from 'routes/utils';
+import { Template, Workspace } from 'types';
+import handleError from 'utils/error';
+
+interface Props {
+  template?: Template;
+  workspaces: Workspace[];
+}
+
+const TemplateViewModalComponent: React.FC<Props> = ({ template, workspaces }) => {
+  const workspace = useMemo(() => {
+    if (!template || !workspaces) return undefined;
+    return workspaces.find((w) => w.id === template.workspaceId);
+  }, [workspaces, template]);
+
+  if (!template || !workspaces) return null;
+
+  return (
+    <Modal size="small" title={`Template ${template.name}`}>
```

Comment:
**nit style:** seems pretty narrow for the code editor, but might just be me.
<img width="1377" alt="Screenshot 2024-05-06 at 4 18 47 PM" src="https://github.com/determined-ai/determined/assets/220971/fbcd5978-d939-4ade-970e-8bccf8f468c0">

Reply:
Good point, changed the modal size to medium and it does look better
<img width="1615" alt="Screenshot 2024-05-07 at 11 02 51 AM" src="https://github.com/determined-ai/determined/assets/40620519/71cac939-45a0-4ed8-a320-e19f241c6795">